packages/hint-create-element-svg/src/hint.ts

```diff
@@ -0,0 +1,96 @@
+const debug: debug.IDebugger = d(__filename);
+
+export default class CreateElementSvgHint implements IHint {
+
+    public static readonly meta = meta;
+
+    /** Finds the approximative location in the page's HTML for a match in an element. */
+    public findProblemLocation(element: IAsyncHTMLElement, content?: string): Promise<ProblemLocation> {
+        return findProblemLocation(element, { column: 0, line: 0 }, content);
+    }
+
+    public constructor(context: HintContext<ScriptEvents>) {
+        const linter = new Linter();
+
+        /*
+         * Check to see if any call expression in the AST has a callee with
+         * a 'createElement' property. If yes, check if that call expression's
+         * first argument is an SVG element. In that case, report an error as
+         * the correct identifier property for creating SVG elements is
+         * 'createElementNS'
+         */
+        linter.defineRule('svg-create', {
+            create(eslintContext) {
+                return {
+                    CallExpression(node) {
+                        const exp = node as ESTree.CallExpression;
+                        const args = exp.arguments;
+                        const callee = exp.callee as ESTree.MemberExpression;
+                        const property = callee !== undefined && callee !== null ? callee.property as ESTree.Identifier : null;
+
+                        if (property !== null && property.name === 'createElement' && args.length >= 1 && args[0].type === 'Literal') {
+                            const arg = args[0] as ESTree.Literal;
+
+                            if (arg.value !== undefined && arg.value !== null && svgElements.has(arg.value.toString().toLowerCase())) {
+                                eslintContext.report({
+                                    messageId: 'avoidElement',
+                                    node: property
+                                });
+                            }
+                        }
+                    }
+                };
+            },
+            meta: { messages: { avoidElement: 'Avoid using createElement to create SVG elements; use createElementNS instead' } }
+        });
+
+        let scriptElement: IAsyncHTMLElement;
+
+        const validateScript = async (scriptData: ScriptParse) => {
+
+            debug(`Validating hint create-element-svg`);
+
+            const sourceCode = scriptData.sourceCode;
+            const results = linter.verify(sourceCode, { rules: { 'svg-create': 'error' } });
+
+            for (const result of results) {
+                let position: ProblemLocation | null = scriptElement.getLocation() || null;
+                let line = 0;
+
+                /*
+                 * Eslint returns location data starting at <script>
+                 * This offsets the line number so that it is relative
+                 * to the full source code and not just the <script> tag
+                 */
+                if (scriptElement) {
```

Comment:
What happens if we encounter a script element referencing an external JavaScript file? We don't want an offset in that case... Probably a good idea to also check that the resource is internal (inline).
```suggestion
                if (scriptElement && scriptData.resource === 'Internal javascript') {
```

Reply:
@molant I was thinking we should export a constant with the `''Internal javascript'` value from `parser-javascript` for just this kind of check. What do you think?

Reply:
The less magic strings the better :)
Maybe as part of a different PR?

Reply:
Agreed on both points.

I was thinking a bit more about this and perhaps we should include an `IAsyncHTMLElement` as part of the `ScriptParse` event when the script was inline? That way we also have the context we need to get the location without having to tie things together across two events (eliminating issues with cached state, etc.)

Reply:
Sounds good to me, although hopefully it will be an `IHTMLElement` (do we need the interface now that everyone should be using the same parser?). Want to open a PR for that?

Reply:
Opened as #1953.